src/main/java/techreborn/compat/libdp/TRTweaker.java

```diff
@@ -0,0 +1,634 @@
+/*
+ * This file is part of TechReborn, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) 2020 TechReborn
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package techreborn.compat.libdp;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonPrimitive;
+import io.github.cottonmc.libdp.api.driver.Driver;
+import io.github.cottonmc.libdp.api.driver.recipe.RecipeDriver;
+import io.github.cottonmc.libdp.api.driver.recipe.RecipeParser;
+import io.github.cottonmc.libdp.api.driver.util.DriverUtils;
+import net.minecraft.fluid.Fluid;
+import net.minecraft.fluid.Fluids;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.recipe.Ingredient;
+import net.minecraft.recipe.ShapedRecipe;
+import net.minecraft.resource.ResourceManager;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.collection.DefaultedList;
+import net.minecraft.util.registry.Registry;
+import reborncore.common.crafting.RebornRecipe;
+import reborncore.common.crafting.RebornRecipeType;
+import reborncore.common.crafting.RecipeManager;
+import reborncore.common.crafting.ingredient.FluidIngredient;
+import reborncore.common.crafting.ingredient.RebornIngredient;
+import reborncore.common.fluid.container.FluidInstance;
+import techreborn.api.generator.EFluidGenerator;
+import techreborn.api.generator.FluidGeneratorRecipe;
+import techreborn.api.generator.GeneratorRecipeHelper;
+import techreborn.api.recipe.recipes.*;
+import techreborn.init.ModRecipes;
+import techreborn.init.TRContent;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.Executor;
+
+public class TRTweaker implements Driver {
```

Comment:
Would better to rename to LibDP's new name scheme (`TRTweaker` -> `TRDriver`) but not mandatory